### ISSUE TITLE
Uncaught exception when node_modules includes a .mjs file

### ISSUE THREAD

**Issue by ggoodman** [wessberg/rollup-plugin-ts#106]
- Version: `1.2.34`
- Rollup Version: `2.23.0`
- Typescript Version: `3.9.7`
- Operating System and version (if applicable): N/A
- Node Version (if applicable): N/A
- Does it work with `tsc` (if applicable): Yes

### Reproduction

1. Install a node module that includes files with the .mjs extension such as `svelte` or `acorn`.
```
npm i svelte
```

2. Attempt to reference such a node module by its bare module specificer.

```ts
import 'svelte';
```

3. Attempt to generate a bundle.

### Expected Behavior

A bundle is generated without error.

### Actual Behavior

```
> rollup -c -w

rollup v2.23.0
bundles src/main.ts → public/build...
[!] (plugin Typescript) TypeError: Cannot read property 'flags' of undefined
node_modules/svelte/index.mjs
TypeError: Cannot read property 'flags' of undefined
    at resolveAlias (<REDACTED>/node_modules/typescript/lib/typescript.js:37340:37)
    at checkAliasSymbol (<REDACTED>/node_modules/typescript/lib/typescript.js:66174:26)
    at checkExportSpecifier (<REDACTED>/node_modules/typescript/lib/typescript.js:66348:13)
    at Object.forEach (<REDACTED>/node_modules/typescript/lib/typescript.js:317:30)
    at checkExportDeclaration (<REDACTED>/node_modules/typescript/lib/typescript.js:66286:24)
    at checkSourceElementWorker (<REDACTED>/node_modules/typescript/lib/typescript.js:66642:28)
    at checkSourceElement (<REDACTED>/node_modules/typescript/lib/typescript.js:66473:17)
    at Object.forEach (<REDACTED>/node_modules/typescript/lib/typescript.js:317:30)
    at checkSourceFileWorker (<REDACTED>/node_modules/typescript/lib/typescript.js:66810:20)
    at checkSourceFile (<REDACTED>/node_modules/typescript/lib/typescript.js:66778:13)
```

**Comment by wessberg** [comment]
Hey there. Thanks for submitting this issue. However, I can't reproduce it easily with a clean project.
I've made this attempt at a repro: https://repl.it/repls/HomelyOldActiveserverpages. Please take a look at it and see what changes to the configuration that needs to be made in order to run into the error. Or, alternatively, prepare a small repro where it happens :-)

**Comment by ggoodman** [comment]
I too had no success reproducing in repli.it. I tried spinning up a monorepo but it seems to have broken repl.it. 🤔

**Comment by ggoodman** [comment]
Here is a [repro](https://repl.it/repls/OnlyWrongAutoresponder#rollup.config.js)

**Comment by ggoodman** [comment]
I isolated it to `"allowJs": true` and the use of `@rollup/plugin-node-resolve`. This appears to cause files to run through this plugin despite:

```
  "exclude": ["**/node_modules/**/*"],
```

**Comment by ggoodman** [comment]
Further, If I add `exclude: /node_modules/` to this plugin, the bug is avoided 🤔 . I wonder if this is by design or a subtle difference between rollup's and typescripts exclusion mechanisms.

**Comment by wessberg** [comment]
Thanks for the update! Much appreciated. So, I found the cause of the issue. I simply thought TypeScript supported `.mjs` files, but [it doesn't](https://github.com/microsoft/TypeScript/issues/27957). I thought the `allowJs` flag included and enabled type checking for `.mjs` files, but it doesn't.

 I've fixed it in [v1.3.1](https://github.com/wessberg/rollup-plugin-ts/releases/tag/v1.3.1) and aligned with the behavior of `tsc`.
I've also written you a message on Twitter about the other issue you mentioned :-) It has been fixed, but there are two small fixes that needs to be made in your monorepro to avoid type duplication :-)
